### PR TITLE
Expose certain parameters to config

### PR DESCRIPTION
**Pull request by chris-langfield**
- `params.skip_processing` for already preprocessed data
- Use `params.data_dtype` in more places to handle non-int16 raw data (e.g. float16)
- `params.skip_unwhiten_before_drift`, turn off inverse whitening transform before drift registration with `dredge`

### CODE REVIEW

**Comment by chris-langfield** [comment]
Tests and integration test pass on server with 

`sudo ./.venv/bin/python examples/integration_100s.py server` from repo directory.

File: examples/integration_100s.py
Comment (oliche) [inline examples/integration_100s.py:1]: Yes we'll do better than that but it's good enough for now !

File: iblsorter/main.py [hunk 1]
```diff
@@ -74,10 +78,20 @@ def run(
 
     ctx = Context(ctx_path)
     ctx.params = params
+    ctx.motion_params = motion_params
```
Comment (oliche) [inline iblsorter/main.py:81]: Here I probably would have added the new parameters groups in a Nested structure. 
This is possible with Pydantic and we would have something like:

params:
- ... old params
- preprocessing
- registration

File: iblsorter/params.py
Comment (oliche) [inline iblsorter/params.py:1]: Yes this is exactly the spirit of nested parameters ! :smiley:

**Review by oliche — CHANGES_REQUESTED** [review]
This is good, however we probably need to add the parameters for the bad channel detection right ? 
I can help with that and keep committing to this branch

**Pull request merged**